packages/lodestar/src/network/network.ts

```diff
@@ -96,11 +96,11 @@ export class Libp2pNetwork extends (EventEmitter as {new (): NetworkEventEmitter
   public async stop(): Promise<void> {
     this.libp2p.connectionManager.removeListener(NetworkEvent.peerConnect, this.emitPeerConnect);
     this.libp2p.connectionManager.removeListener(NetworkEvent.peerDisconnect, this.emitPeerDisconnect);
+    await Promise.all([this.diversifyPeersTask.stop(), this.checkPeerAliveTask.stop()]);
```

Comment:
everything else in the PR looks good to me, but i'm curious, what it the effect of stopping these tasks here instead of where it used to be?  What motivated this change?

Reply:
These tasks are for searching for peers and pinging peers. Before, we were shutting these down _after_ we shut down libp2p, and no longer had any peers. So there was a slight change that tasks could error or not shut down cleanly.

This was usually not even visible or a practical issue, since these tasks run on long intervals, and would shut down before having a chance to run after libp2p had been shut down. During debugging this PR, this little wrinkle became apparent tho.